### ISSUE TITLE
Ajax request for external sprite with hash

### ISSUE THREAD

**Issue by thomasaull** [kisenka/svg-sprite-loader#176]
I've got a question: In my current setup, I'm usinkg svg-sprite-loader to extract all svgs in an external file, which I'm loading in my website via an ajax-request. However, if I try to use a hashed filename, I have no idea how to build my request URL. I tried to use the webpack.ExtendedAPIPlugin which publishes a __webpack_hash__ variable, but some this differs from the hash generated by svg-sprite-loader

Relevant part of my webpack.config.js:
```
{
  test: /\.svg$/,
  use: [
    { loader: 'svg-sprite-loader', options: {
      extract: true,
      spriteFilename: 'sprite.[hash].svg'
    }},
    { loader: 'svgo-loader', options: {
      plugins: svgoplugins
    }}
  ]
}
```

**Comment by kisenka** [comment]
@thomasaull and that is the question? You need to determine hash of sprite file?

**Comment by thomasaull** [comment]
Yes, basically I need the hash in one of my javascript files

**Comment by kisenka** [comment]
It can be done via simple plugin which find sprite file in assets and replace some placeholder in sources with it.

**app.js**
```js
const spriteUrl = '__SPRITE_URL__'; // this placeholder will be replaced with actual sprite filename
```

**webpack.config.js**
```js
const { replaceInModuleSource, getAllModules } = require('svg-sprite-loader/lib/utils');

plugins: [
  ...,

  {
      apply(compiler) {
        compiler.plugin('emit', (compilation, done) => {
          const { assets } = compilation;
          const spriteFilename = Object.keys(assets)
            .find(assetName => assetName.startsWith('sprite.'));

          getAllModules(compilation).forEach((module) => {
            replaceInModuleSource(module, {
              __SPRITE_URL__: spriteFilename
            });
          });

          done();
        });
      }
    }
]
```

**Comment by kisenka** [comment]
@thomasaull ping

**Comment by thomasaull** [comment]
I'm sorry Stas, I was quite busy the last days, I'm gonna try your suggestion next week! :)

**Comment by thomasaull** [comment]
No… Unfortunately this doesn't work. If i console.log my spriteUrl it is still the string '__SPRITE_URL__' … @kisenka

**Comment by kisenka** [comment]
@thomasaull I need a test repo then, It's fastest way to help you

**Comment by kisenka** [comment]
@thomasaull ping, still actual?

**Comment by thomasaull** [comment]
@kisenka Kinda… But not urgent right now. If ok for you, I'm gonna reopen this issue as soon as I got a test repo for you!

**Comment by kisenka** [comment]
@thomasaull have you tried solution in my [comment](https://github.com/kisenka/svg-sprite-loader/issues/176#issuecomment-322622821). Problem is that there could be a few sprite files and simple replace placeholder with hash will not work.

**Comment by thomasaull** [comment]
I did try that and the result was, that somehow the __SPRITE_URL__ wasn't replaced with the actual sprite Url, it was still just __SPRITE_URL__ in my compiled code…

**Comment by thomasaull** [comment]
@kisenka I spend some hours trying to figure out what's wrong… turn's out the Uglify Plugin interferes in some way that the `__SPRITE_URL__` replacement does not work anymore. You can find a test repo over here: https://github.com/thomasaull/svg-sprite-loader_testrepo

if you remove the UglifyJS Plugin at the bottom of the webpack.conifg and run `npm run dev` it's working. With UglifyJS it's not… Would be awesome if you could look into this, since my webpack knowledge is too limited to figure this one out…

**Comment by kisenka** [comment]
@thomasaull sorry for the delay, I'll check your demo ASAP

**Comment by thomasaull** [comment]
Any news @kisenka?

**Comment by RehanSaeed** [comment]
Also interested in how to get the URL of the icon sprite file containing the hash.

**Comment by pdoreau** [comment]
Me too. I also the tried the `__SPRITE_URL__` solution and this deprecation message is displayed : 

`(node:3601) DeprecationWarning: Tapable.plugin is deprecated. Use new API on .hooks instead`

Did you guys finally make it work ?
@kisenka do you now recommend something else ?
